libs/vkd3d/device.c

```diff
@@ -3412,29 +3847,105 @@ static D3D12_RESOURCE_BINDING_TIER d3d12_device_determine_resource_binding_tier(
-    caps->options.TypedUAVLoadAdditionalFormats = features->shaderStorageImageExtendedFormats;
+    options->PSSpecifiedStencilRefSupported = FALSE;
+    options->TypedUAVLoadAdditionalFormats = features->shaderStorageImageExtendedFormats;
     /* Requires VK_EXT_fragment_shader_interlock */
-    caps->options.ROVsSupported = FALSE;
+    options->ROVsSupported = FALSE;
     /* Requires VK_EXT_conservative_rasterization */
-    caps->options.ConservativeRasterizationTier = D3D12_CONSERVATIVE_RASTERIZATION_TIER_NOT_SUPPORTED;
-    caps->options.MaxGPUVirtualAddressBitsPerResource = 40; /* XXX */
-    caps->options.StandardSwizzle64KBSupported = FALSE;
-    caps->options.CrossNodeSharingTier = D3D12_CROSS_NODE_SHARING_TIER_NOT_SUPPORTED;
-    caps->options.CrossAdapterRowMajorTextureSupported = FALSE;
+    options->ConservativeRasterizationTier = D3D12_CONSERVATIVE_RASTERIZATION_TIER_NOT_SUPPORTED;
+    options->MaxGPUVirtualAddressBitsPerResource = 40; /* XXX */
+    options->StandardSwizzle64KBSupported = FALSE;
+    options->CrossNodeSharingTier = D3D12_CROSS_NODE_SHARING_TIER_NOT_SUPPORTED;
+    options->CrossAdapterRowMajorTextureSupported = FALSE;
     /* Requires VK_EXT_shader_viewport_index_layer */
-    caps->options.VPAndRTArrayIndexFromAnyShaderFeedingRasterizerSupportedWithoutGSEmulation = FALSE;
-    caps->options.ResourceHeapTier = D3D12_RESOURCE_HEAP_TIER_2;
+    options->VPAndRTArrayIndexFromAnyShaderFeedingRasterizerSupportedWithoutGSEmulation = FALSE;
+    options->ResourceHeapTier = D3D12_RESOURCE_HEAP_TIER_2;
+}
+
+static void d3d12_device_caps_init_feature_options1(struct d3d12_device *device)
+{
+    D3D12_FEATURE_DATA_D3D12_OPTIONS1 *options1 = &device->d3d12_caps.options1;
+
+    options1->WaveOps = device->d3d12_caps.max_shader_model >= D3D_SHADER_MODEL_6_0;
+
+    if (device->vk_info.EXT_subgroup_size_control)
+    {
+        options1->WaveLaneCountMin = device->device_info.subgroup_size_control_properties.minSubgroupSize;
+        options1->WaveLaneCountMax = device->device_info.subgroup_size_control_properties.maxSubgroupSize;
+    }
+    else
+    {
+        WARN("Device info for WaveLaneCountMin and WaveLaneCountMax may be inaccurate.\n");
+        options1->WaveLaneCountMin = device->device_info.subgroup_properties.subgroupSize;
+        options1->WaveLaneCountMax = device->device_info.subgroup_properties.subgroupSize;
+    }
+
+    if (device->vk_info.AMD_shader_core_properties)
+    {
+        const VkPhysicalDeviceShaderCorePropertiesAMD *amd = &device->device_info.shader_core_properties;
+        const VkPhysicalDeviceShaderCoreProperties2AMD *amd2 = &device->device_info.shader_core_properties2;
+        uint32_t compute_units;
+        
+        if (device->vk_info.AMD_shader_core_properties2)
+            compute_units = amd2->activeComputeUnitCount;
+        else
+            compute_units = amd->shaderEngineCount *
+                    amd->computeUnitsPerShaderArray *
+                    amd->wavefrontSize;
```

Comment:
Is wavefrontSize 64 here?

Reply:
Ah whoops, this shouldn't even be in here. Derp from rebasing.